src/audio/drc/drc.c

```diff
@@ -370,7 +370,7 @@ static int drc_reset(struct processing_module *mod)
 
 	drc_reset_state(&cd->state);
 
-	cd->source_format = 0;
+	cd->sink_format = 0;
```

Comment:
this is `SOF_IPC_FRAME_S16_LE` - is this what we want?

Reply:
Component private data is rzalloc'ed so this way the component is reset to init state. I think use of zero here instead of enum is correct. Do you agree?

Reply:
I would just delete the line or use the enum, leaving as it is not helpful

Reply:
> Component private data is rzalloc'ed so this way the component is reset to init state. I think use of zero here instead of enum is correct. Do you agree?

@singalsu no, actually I don't. Then I'd say the initialisation is also wrong - if that field is explicitly initialised to 0 there. It is ok to leave objects uninitialised *until* they're first used - then of course they must be initialised. I think you have two possibilities:
1. you initialise an object when it's first instantiated, then whenever you get to use it you know, that it is in some consistent state
2. you initialise data any time *before* the first use. In this case it's find to leave that data in some random undefined state until then, but of course then you must know exactly *when* your data is needed for the first time.

Both ways are valid. The second one can be more optimal in terms of performance - you don't "waste" cycles on initialisation, if it's anyway going to be overwritten later, but it might also be more complex - you risk missing the point when the data is used and you end up consuming random data.

Either way when you do set your data to something, it should be something *meaningful*, and not 0 just because it's some magic universal initialiser :-)

Reply:
Fixed now, deleted the line.